cmd/soroban-rpc/internal/events/events.go

```diff
@@ -207,31 +163,47 @@ func seek(events []event, cursor Cursor) []event {
 // IngestEvents adds new events from the given ledger into the store.
 // As a side effect, events which fall outside the retention window are
 // removed from the store.
-func (m *MemoryStore) IngestEvents(txReader *ingest.LedgerTransactionReader) error {
-	events, err := readEvents(txReader)
+func (m *MemoryStore) IngestEvents(ledgerCloseMeta xdr.LedgerCloseMeta) error {
+	// no need to acquire the lock because the networkPassphrase field
+	// is immutable
+	events, err := readEvents(m.networkPassphrase, ledgerCloseMeta)
 	if err != nil {
 		return err
 	}
-	ledgerSequence := txReader.GetSequence()
-	return m.append(ledgerSequence, events)
+	ledgerSequence := ledgerCloseMeta.LedgerSequence()
+	ledgerCloseTime := int64(ledgerCloseMeta.LedgerHeaderHistoryEntry().Header.ScpValue.CloseTime)
+	return m.append(ledgerSequence, ledgerCloseTime, events)
 }
 
-func readEvents(txReader *ingest.LedgerTransactionReader) ([]event, error) {
-	var events []event
+func readEvents(networkPassphrase string, ledgerCloseMeta xdr.LedgerCloseMeta) (events []event, err error) {
+	var txReader *ingest.LedgerTransactionReader
+	txReader, err = ingest.NewLedgerTransactionReaderFromLedgerCloseMeta(networkPassphrase, ledgerCloseMeta)
+	if err != nil {
+		return
+	}
+	defer func() {
+		err = txReader.Close()
```

Comment:
this is the "tricky" portion I mentioned in my comment...
I think that what you want in the following:
```golang
defer func() {
		closeErr := txReader.Close()
		if err == nil {
			err = closeErr
		}
}
```